java-generator/core/src/main/java/io/fabric8/java/generator/nodes/JObject.java

```diff
@@ -0,0 +1,250 @@
+                                new VariableDeclarator()
+                                        .setName(ADDITIONAL_PROPERTIES)
+                                        .setType(mapType)
+                                        .setInitializer(
+                                                "new java.util.HashMap<String, Object>()")));
+
+                objField.addAnnotation("com.fasterxml.jackson.annotation.JsonIgnore");
+
+                objField.createGetter()
+                        .addAnnotation("com.fasterxml.jackson.annotation.JsonAnyGetter");
+                objField.createSetter()
+                        .addAnnotation("com.fasterxml.jackson.annotation.JsonAnySetter");
+            } else {
+                // Warning ???
+            }
+        }
+
+        List<String> buffer = new ArrayList<>(this.fields.size() + 1);
+
+        // CU to expand inner Enums
+        CompilationUnit supportCU = new CompilationUnit();
+        for (String k : this.fields.keySet()) {
+            AbstractJSONSchema2Pojo prop = this.fields.get(k);
+            boolean isRequired = this.required.contains(k);
+
+            GeneratorResult gr = prop.generateJava(supportCU);
+
+            // For now the inner types are only for enums
+            if (!gr.getInnerClasses().isEmpty()) {
+                for (String enumName : gr.getInnerClasses()) {
+                    Optional<EnumDeclaration> ed = supportCU.getEnumByName(enumName);
+                    if (ed.isPresent()) {
+                        clz.addMember(ed.get());
+                    }
+                }
+            }
+
+            gr = prop.generateJava(cu);
+            buffer.addAll(gr.getTopLevelClasses());
+
+            String originalFieldName = k;
+            String fieldName = AbstractJSONSchema2Pojo.sanitizeString(k);
+            String fieldType = AbstractJSONSchema2Pojo.sanitizeString(prop.getType());
+
+            if (!clz.getFieldByName(fieldName).isPresent()) {
+                try {
+                    FieldDeclaration objField =
+                            clz.addField(fieldType, fieldName, Modifier.Keyword.PRIVATE);
+                    objField.addAnnotation(
+                            new SingleMemberAnnotationExpr(
+                                    new Name("com.fasterxml.jackson.annotation.JsonProperty"),
+                                    new StringLiteralExpr(originalFieldName)));
+
+                    if (isRequired) {
+                        objField.addAnnotation("javax.validation.constraints.NotNull");
+                    }
+
+                    objField.createGetter();
+                    objField.createSetter();
+                } catch (Exception cause) {
+                    throw new RuntimeException(
+                            "Error generating field " + fieldName + " with type " + prop.getType(),
+                            cause);
+                }
+            } else {
+                // Warning ???
```

Comment:
At least IMHO it should definitely be logged.

Reply:
I have the code to remove the corresponding `if` along with the warning above, this should not happen at all.

Reply:
As stated previously, it is ok to me if this will be tracked as a future improvement, thanks!